test/cli/spdx_tooling_validation_test.go

```diff
@@ -0,0 +1,85 @@
+package cli
+
+import (
+	"fmt"
+	"log"
+	"os"
+	"os/exec"
+	"path"
+	"path/filepath"
+	"strings"
+	"testing"
+)
+
+func TestSpdxValidationTooling(t *testing.T) {
+	tests := []struct {
+		name       string
+		syftArgs   []string
+		images     []string
+		env        map[string]string
+		assertions []traitAssertion
+	}{
+		{
+			name:     "spdx validation tooling tag value",
+			syftArgs: []string{"packages", "-o", "spdx"},
+			images:   []string{"alpine:latest", "photon:3.0", "debian:latest"},
+			env: map[string]string{
+				"SYFT_FILE_METADATA_CATALOGER_ENABLED": "true",
+				"SYFT_FILE_METADATA_DIGESTS":           "sha1",
+			},
+			assertions: []traitAssertion{
+				assertSuccessfulReturnCode,
+			},
+		},
+	}
+
+	for _, test := range tests {
+		t.Run(test.name, func(t *testing.T) {
+			for _, image := range test.images {
+				args := append(test.syftArgs, image)
+				cmd, stdout, stderr := runSyft(t, test.env, args...)
+				for _, traitFn := range test.assertions {
+					traitFn(t, stdout, stderr, cmd.ProcessState.ExitCode())
+				}
+
+				cwd, err := os.Getwd()
+				if err != nil {
+					t.Fatalf("failed to get working directory: %+v", err)
+				}
+
+				var rename string
+				if strings.Contains(test.name, "json") {
+					rename = "test.json"
+				} else {
+					rename = "test.spdx"
+				}
+
+				f, err := os.CreateTemp(path.Join(cwd, "test-fixtures", "image-java-spdx-tools"), rename)
```

Comment:
this is creating a temp directory within the project tree? In case anything goes wrong, I'd feel better about having temp dirs being created in the OS standard spot... otherwise we could have temp dirs that end up in the project tree and accidentally get checked in or accumulated over time.

Reply:
Yea let me take a second pass here.

This is creating a temporary file. That file is then mounted into the docker container with the spdx tooling. I would also prefer this probably to exist at the OS standard spot - I think I can get the mount to work that way rather than write the file to this exact directory

Reply:
related nit: Also, I see why `os.CreateTemp` (file) is used over `t.TempDir`, though the upside of the latter is that cleanup is done for free (though it does take one extra `os.Open` to use... so somewhat of a tradeoff).

Reply:
moved this so that `os.CreateTemp` uses t.TempDir so we get the free cleanup